crates/orderbook/src/database/solver_competition.rs

```diff
@@ -47,6 +51,36 @@ impl SolverCompetitionStoring for Postgres {
             .context("order_rewards::save")?;
         }
 
+        database::settlement_scores::insert(
+            &mut ex,
```

Comment:
Okay at this point it seems clear that the DB insertions actually should take a `PgTransaction` instead of starting their own. 😅

Reply:
Fixed.

Reply:
I think inserting `scores` should also be made part of a DB transaction now. This was not really clear during the previous review.